### PR TITLE
fix(hyperliquid): fetchOHLCV - timestamp for candle is opening timestamp

### PR DESCRIPTION
**Pull request by samgermain**
fixes: #21783

# Testing

```
% py hyperliquid fetchOHLCV BTC/USDC:USDC | condense
Python v3.9.6
CCXT v4.2.76
hyperliquid.fetchOHLCV(BTC/USDC:USDC)
[[1710552360000, 69584.0, 69610.0, 69474.0, 69483.0, 6.83271],
 [1710552420000, 69471.0, 69497.0, 69465.0, 69497.0, 0.88301],
...
 [1710582180000, 69165.0, 69165.0, 69096.0, 69098.0, 1.37846],
 [1710582240000, 69098.0, 69098.0, 69066.0, 69070.0, 0.49219],
 [1710582300000, 69059.0, 69060.0, 68978.0, 68986.0, 1.43576]]
```

```
% py binance fetchOHLCV ETH/USDT:USDT-240329-600.00000000-C | condense   
Python v3.9.6
CCXT v4.2.76
binance.fetchOHLCV(ETH/USDT:USDT-240329-600.00000000-C)
[[1710828000000, 2666.0, 2666.0, 2666.0, 2666.0, 0.0],
 [1710828060000, 2666.0, 2666.0, 2666.0, 2666.0, 0.0],
 [1710828120000, 2666.0, 2666.0, 2666.0, 2666.0, 0.0],
...
 [1710857820000, 2666.0, 2666.0, 2666.0, 2666.0, 0.0],
 [1710857880000, 2666.0, 2666.0, 2666.0, 2666.0, 0.0],
 [1710857940000, 2666.0, 2666.0, 2666.0, 2666.0, 0.0]]
```

### CODE REVIEW

File: ts/src/hyperliquid.ts [hunk 1]
```diff
@@ -570,7 +570,7 @@ export default class hyperliquid extends Exchange {
         //     }
         //
         return [
-            this.safeInteger (ohlcv, 'T'),
+            this.safeInteger (ohlcv, 't'),
```
Comment (carlosmiei) [inline ts/src/hyperliquid.ts:573]: @samgermain I'm not sure this is correct 
image

the opening time of the candle can be different from the first trade time I think

Comment (samgermain) [reply ts/src/hyperliquid.ts:573]: I separated the binance changes into a separate commit because of this delay https://github.com/ccxt/ccxt/pull/21811

Comment (samgermain) [reply ts/src/hyperliquid.ts:573]: @carlosmiei I think that's just bad documentation on their part, the timestamps fall exactly on the minute time periods and are evenly spaced

```
% date -r 1710552360   
Fri Mar 15 21:26:00 EDT 2024
% date -r 1710552420   
Fri Mar 15 21:27:00 EDT 2024
% date -r 1710582180    
Sat Mar 16 05:43:00 EDT 2024
% date -r 1710582240    
Sat Mar 16 05:44:00 EDT 2024
% date -r 1710582300   
Sat Mar 16 05:45:00 EDT 2024
```

Comment (carlosmiei) [reply ts/src/hyperliquid.ts:573]: @samgermain yeah just checked that, will merge it soon just waiting for the build (I suspect some response tests might break with this)

**Pull request merged**